src/tbb/concurrent_monitor.h

```diff
@@ -437,7 +437,16 @@ class concurrent_monitor_base {
             my_waitset.flush_to(temp);
             end = temp.end();
             for (base_node* n = temp.front(); n != end; n = n->next) {
+// GCC 12.x-14.x issues a warning here that to_wait_node(n)->my_is_in_list might have size 0, since n is
+// a base_node pointer. (This cannot happen, because only wait_node pointers are added to my_waitset.)
+#if (__TBB_GCC_VERSION >= 120100 && __TBB_GCC_VERSION < 150000 ) && !__clang__ && !__INTEL_COMPILER
```

Comment:
Since the same warning is suppressed on the line 296 and the macros in the "#if" statement are the same, we have the same if statement 4 times. I propose to define the special macro with the content of the statement and reuse it in these 4 places.